sdk/iot/core/src/az_iot_core.c

```diff
@@ -8,74 +8,41 @@
 #include <az_result.h>
 #include <az_span.h>
 
+#include <az_retry_internal.h>
+
 #include <_az_cfg.h>
 
-AZ_NODISCARD az_result az_iot_get_status_from_uint32(uint32_t status_int, az_iot_status* status)
+AZ_NODISCARD int32_t az_iot_retry_calc_delay(
+    int32_t operation_msec,
+    int16_t attempt,
+    int32_t min_retry_delay_msec,
+    int32_t max_retry_delay_msec,
+    int32_t random_msec)
```

Comment:
This is a relatively complex API signature, especially for something that is public. Any way to design it more simply?

Does attempt need to be an `int16_t`, or can we just keep it `int32_t` like the rest?

Reply:
> This is a relatively complex API signature, especially for something that is public. Any way to design it more simply?

I had a different approach that would require state. We all agreed the overheads are too big.

> Does attempt need to be an int16_t, or can we just keep it int32_t like the rest?

It's the same as the internal that Az SDKs proposed and so I can't use a larger int unless the internal also changes.